src/Altinn.Notifications/Validators/SmsNotificationOrderRequestValidator.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Text.RegularExpressions;
+
+using Altinn.Notifications.Models;
+
+using FluentValidation;
+using PhoneNumbers;
+
+namespace Altinn.Notifications.Validators;
+
+/// <summary>
+/// Class containing validation logic for the <see cref="SmsNotificationOrderRequestExt"/> model
+/// </summary>
+public class SmsNotificationOrderRequestValidator : AbstractValidator<SmsNotificationOrderRequestExt>
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SmsNotificationOrderRequestValidator"/> class.
+    /// </summary>
+    public SmsNotificationOrderRequestValidator()
+    {
+        RuleFor(order => order.Recipients)
+            .NotEmpty()
+            .WithMessage("One or more recipient is required.")
+            .Must(recipients => recipients.TrueForAll(a => IsValidMobileNumber(a.MobileNumber)))
+            .WithMessage("A valid mobile number must be provided for all recipients.");
+
+        RuleFor(order => order.RequestedSendTime)
+            .Must(sendTime => sendTime.Kind != DateTimeKind.Unspecified)
+            .WithMessage("The requested send time value must have specified a time zone.")
+            .Must(sendTime => sendTime >= DateTime.UtcNow.AddMinutes(-5))
+            .WithMessage("Send time must be in the future. Leave blank to send immediately.");
+
+        RuleFor(order => order.Body).NotEmpty();
+    }
+
+    /// <summary>
+    /// Validated as mobile number based on the Altinn 2 regex
+    /// </summary>
+    /// <param name="mobileNumber">The string to validate as an mobile number</param>
+    /// <returns>A boolean indicating that the mobile number is valid or not</returns>
+    internal static bool IsValidMobileNumber(string? mobileNumber)
+    {
+        if (string.IsNullOrEmpty(mobileNumber))
+        {
+            return false;
+        }
+
+        string mobileNumberRegexPattern = @"^(?:(\+47[49]\d{7})|(0047[49]\d{7})|(?!((\+47[0-9]*)|(0047[0-9]*)))(([0-9]{8})|(00[0-9]{3,})|(\+[0-9]{3,})))$";
+
+        Regex regex = new(mobileNumberRegexPattern, RegexOptions.None, TimeSpan.FromSeconds(1));
+
+        Match match = regex.Match(mobileNumber);
+
+        if (!match.Success)
+        {
+            return false;
+        }
+
+        PhoneNumberUtil phoneNumberUtil = PhoneNumberUtil.GetInstance();
+
+        if (!mobileNumber.StartsWith('+'))
+        {
+            if (mobileNumber.StartsWith("00"))
+            {
+                mobileNumber = "+" + mobileNumber.Remove(0, 2);
+            }
+            else
+            {
+                mobileNumber = "+47" + mobileNumber;
+            }
+        }
+
+        PhoneNumber phoneNumber = phoneNumberUtil.Parse(mobileNumber, null);
+        return phoneNumberUtil.IsValidNumber(phoneNumber);
```

Comment:
would the phoneNumberUtil catch most cases that our regex also catches? Wondering if we could call that first, and then only validate the norwegian number starting with 4 or 9 in the later check? But that depends on how thorough the Util class is.

Reply:
Unresolving, think this is still relevant. Will this cover everything the regex covers ?

Reply:
After we decided to require + or 00 in the api I think the regex validation is unnecessary. I tested very little, but the library should validate against norwegian rules. I tried with a number starting on 5 and that failed. 
Test the library enough to see that it covers what we validates in the regex, then we could drop the regex.

Reply:
I also think the RegEx is unnecessary now as the library is pretty smart! So just using the library can achieve what we are doing now I guess. 🤔

Reply:
So far all the checks has been passed! So we will be relying on the library from now on and dropping the RegEx. :)